build-go.sh

```diff
@@ -324,11 +327,11 @@ function startGoBuilds() {
 	fi
 
 	if [ -z "$existing" ]; then
-		existing="/ipns/dist.ipfs.io"
+		existing="$DIST_PATH"
 	fi
 
 	echo "comparing $versions with $existing/$distname/versions"
-	newVersions=$(comm -13 <(ipfs cat "$existing/$distname/versions") "$versions")
+	newVersions=$(comm --nocheck-order -13 <(ipfs cat "$existing/$distname/versions") "$versions")
```

Comment:
This fixes an issue when trying to build a distribution that needs semver sorting.  Without the `--nocheck-order` flag, sorting breaks the diff check because the following:
```
v0.1.0
v0.2.0
v0.10.0
```
gets incorrectly reordered as:
```
v0.1.0
v0.10.0
v0.2.0
```
Without this fix, you cannot build https://dist.ipfs.io/ipfs-cluster-service